src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/Navigation.java

```diff
@@ -350,21 +350,28 @@ private void addMainNavLinks(Content tree) {
      * Adds the summary links to the sub-navigation.
      *
      * @param tree the content tree to which the sub-navigation will added
+     * @param nested whether to create a flat or nested list
      */
-    private void addSummaryLinks(Content tree) {
+    private void addSummaryLinks(Content tree, boolean nested) {
         switch (documentedPage) {
             case MODULE, PACKAGE, CLASS, HELP -> {
                 List<? extends Content> listContents = subNavLinks.getSubNavLinks()
                         .stream().map(HtmlTree::LI).toList();
                 if (!listContents.isEmpty()) {
-                    tree.add(HtmlTree.LI(switch (documentedPage) {
+                Content label = switch (documentedPage) {
                         case MODULE -> contents.moduleSubNavLabel;
                         case PACKAGE -> contents.packageSubNavLabel;
                         case CLASS -> contents.summaryLabel;
                         case HELP -> contents.helpSubNavLabel;
                         default -> Text.EMPTY;
-                    }).add(Entity.NO_BREAK_SPACE));
-                    addListToNav(listContents, tree);
+                    };
+                    if (nested) {
+                        tree.add(HtmlTree.LI(HtmlTree.P(label))
```

Comment:
Is the `HtmlTree.P` necessary/useful?

Reply:
I think it is necessary, at least I haven't found an alternative solution. 

The `<li>` element we are creating here contains the main label as well as a nested list of links. The `<li>` elements of the nested list are sized and padded via CSS. In order to apply similar size and padding to the main label it needs to be contained in an element with block display.

I realize the use of `<p>` instead of `<div>` may seem a bit strange as the label does not really represent a paragraph. I think I chose it because it had more useful default sizing rules. However, I could easily switch to using a `<div>` instead.